docs/2.8.0/docs/daml-finance/tutorials/advanced-topics/instrument-modeling/calendar-dcf.rst

```diff
@@ -0,0 +1,268 @@
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Calendar.daml
+  :language: daml
+  :start-after: -- CREATE_PERIODIC_SCHEDULE_BEGIN
+  :end-before: -- CREATE_PERIODIC_SCHEDULE_END
+
+Since there is one year between the start and the end date, we would expect this periodic schedule
+to correspond to four periods of 3M each:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Calendar.daml
+  :language: daml
+  :start-after: -- CREATE_EXPECTED_SCHEDULE_RESULT_BEGIN
+  :end-before: -- CREATE_EXPECTED_SCHEDULE_RESULT_END
+
+Now, let us roll out this schedule to get the specific start and end date for each period, and
+compare it to the expected result:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Calendar.daml
+  :language: daml
+  :start-after: -- CREATE_SCHEDULE_BEGIN
+  :end-before: -- CREATE_SCHEDULE_END
+
+An important aspect of schedules is the concept of *stub periods*. The above schedule only had
+regular 3M periods. However, if the sample schedule would start one month later:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Calendar.daml
+  :language: daml
+  :start-after: -- CREATE_PERIODIC_SCHEDULE_WITH_STUB_BEGIN
+  :end-before: -- CREATE_PERIODIC_SCHEDULE_WITH_STUB_END
+
+it would only have 11 months in total, which does not match with neither 3 nor 4 regular 3M periods.
+Instead, we need to define a stub period, so that the rest of the schedule is regular.
+In the example above, we did this by defining a start date for the first regular period. This
+implies a short initial stub period, in our case 2 months. We would expect the following schedule:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Calendar.daml
+  :language: daml
+  :start-after: -- CREATE_EXPECTED_SCHEDULE_RESULT_WITH_STUB_BEGIN
+  :end-before: -- CREATE_EXPECTED_SCHEDULE_RESULT_WITH_STUB_END
+
+Different stub types can be configured using the
+:ref:`StubPeriodTypeEnum <type-daml-finance-interface-types-date-schedule-stubperiodtypeenum-69372>`
+data type. You can configure whether the stub period should be at the beginning or at the end of the
+schedule, as well as whether the stub period should be longer or shorter than a regular period.
+
+Let us roll out this schedule, which includes a stub period, and compare it to the expected result:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Calendar.daml
+  :language: daml
+  :start-after: -- CREATE_SCHEDULE_WITH_STUB_BEGIN
+  :end-before: -- CREATE_SCHEDULE_WITH_STUB_END
+
+The schedule can be used to represent different aspects of a financial instrument, for example:
+
+- payment periods, which define e.g. *when* a bond coupon or a swap payment should be paid
+- calculation periods, which define e.g. *how* the interest amount is calculated
+
+DayCount
+========
+
+Many instruments that pay interest require an exact definition of how many days of interest belong
```

Comment:
We should specify that this is relevant as the coupon rate is often expressed in annualized terms